frame/composable-support/src/validation.rs

```diff
@@ -0,0 +1,98 @@
+use core::marker::PhantomData;
+use scale_info::TypeInfo;
+
+/// Black box that embbed the validated value.
+#[derive(Eq, PartialEq, Debug, TypeInfo, codec::Encode)]
+pub struct Validated<T, U> {
+	pub value: T,
+	_marker: PhantomData<U>,
+}
+
+pub trait Validate<U> {
+	fn validate(&self) -> Result<(), &'static str>;
+}
```

Comment:
Changing this to `fn validate(self) -> Result<Self, ...>;` will allow implementors to modify as well, such as performing sorting operations etc.

Reply:
Perhaps also have it return a DispatchError for better substrate compatibility.

Reply:
Or look into how errors are produced, as I think they have an intermediate trait for the covnersion step